pkg/agent/plugin/workloadattestor/k8s/sigstore/sigstorecache_test.go

```diff
@@ -0,0 +1,209 @@
+			{
+
+				Subject:        "spirex1@example.com",
+				Content:        "content1",
+				LogID:          "1111111111111111",
+				IntegratedTime: "1111111111111111",
+			},
+		},
+	}
+
+	selectors2 = Item{
+		Key: "signature2",
+		Value: []SelectorsFromSignatures{
+			{
+
+				Subject:        "spirex2@example.com",
+				Content:        "content2",
+				LogID:          "2222222222222222",
+				IntegratedTime: "2222222222222222",
+			},
+		},
+	}
+
+	selectors3 = Item{
+		Key: "signature3",
+		Value: []SelectorsFromSignatures{
+			{
+				Subject:        "spirex3@example.com",
+				Content:        "content3",
+				LogID:          "3333333333333333",
+				IntegratedTime: "3333333333333333",
+			},
+		},
+	}
+
+	selectors3Updated = Item{
+		Key: "signature3",
+		Value: []SelectorsFromSignatures{
+			{
+				Subject:        "spirex3@example.com",
+				Content:        "content4",
+				LogID:          "4444444444444444",
+				IntegratedTime: "4444444444444444",
+			},
+		},
+	}
+)
+
+func TestNewCache(t *testing.T) {
+	tests := []struct {
+		name string
+		want Cache
+	}{
+		{
+			name: "New",
+			want: &cacheImpl{
+				size:     3,
+				items:    list.New(),
+				mutex:    sync.RWMutex{},
+				itemsMap: make(map[string]MapItem),
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := NewCache(3); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("NewCache() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestCacheimpl_GetSignature(t *testing.T) {
+	m := make(map[string]MapItem)
+	items := list.New()
+
+	m[selectors1.Key] = MapItem{
+		item:    &selectors1,
+		element: items.PushFront(selectors1.Key),
+	}
+	m[selectors2.Key] = MapItem{
+		item:    &selectors2,
+		element: items.PushFront(selectors2.Key),
+	}
+
+	cacheInstance := &cacheImpl{
+		size:     3,
+		items:    items,
+		mutex:    sync.RWMutex{},
+		itemsMap: m,
+	}
+
+	tests := []struct {
+		name         string
+		want         *Item
+		key          string
+		errorMessage string
+	}{
+		{
+			name:         "Non existing entry",
+			want:         nil,
+			key:          selectors3.Key,
+			errorMessage: "A non-existing item's key should return a nil item.",
+		},
+		{
+			name:         "Existing entry",
+			want:         &selectors1,
+			key:          selectors1.Key,
+			errorMessage: "An existing items key's should return the existing item",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := cacheInstance.GetSignature(tt.key); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("%v Got: %v Want: %v", tt.errorMessage, got, tt.want)
+			}
+		})
+	}
+}
+
+func TestCacheimpl_PutSignature(t *testing.T) {
+	m := make(map[string]MapItem)
+	items := list.New()
+
+	cacheInstance := &cacheImpl{
+		size:     2,
+		items:    items,
+		mutex:    sync.RWMutex{},
+		itemsMap: m,
+	}
+
+	tests := []struct {
+		name       string
+		item       *Item
+		wantLength int
+		wantKey    string
+		wantValue  *Item
+	}{
+		{
```

Comment:
this progresive approach using the same cache to test different test cases can result in unexpected behavior, 
all unit test must not depends of previous cases.